src/puppetlabs/trapperkeeper/authorization/rules.clj

```diff
@@ -64,6 +56,18 @@
       (assoc :file file)
       (assoc :line line)))
 
+(schema/defn query-param :- Rule
+  "Add a query parameter matching value(s) to a rule. New values will be
+   appended to existing values.
+
+   The query parameters are in a map under the `:match-request` section of the
+   rule. Keys in the map are strings corresponding to the query parameters to
+   match, and the values are sequences of strings of acceptable values."
+  [rule :- Rule
+   param :- schema/Str
+   value :- (schema/either schema/Str [schema/Str])]
+  (update-in rule [:query-params param] into (flatten [value])))
```

Comment:
Since it looks like we end up converting the flattened value into a set each time we do a `query-params-match?` later, should we just do the conversion to set here?